src/utilities/bcl/BCLXML.cpp

```diff
@@ -727,7 +736,11 @@ pugi::xml_document BCLXML::toXML() const {
 
   // write files
   element = docElement.append_child("files");
-  for (const BCLFileReference& file : m_files) {
+
+  // Fix for #4748 - sort files
+  auto sortedFiles = m_files;
+  std::sort(sortedFiles.begin(), sortedFiles.end());
+  for (const BCLFileReference& file : sortedFiles) {
```

Comment:
I have considered making m_files a std::set so that addFile(BCLFileReference&) would always leave things sorted, but I didn't want to slow down anything. I can revisit if someone feels strongly about the need to have m_files (and the return of BCLXML::files()) sorted always.